src/base/path.lua

```diff
@@ -142,24 +142,53 @@
 
 
 --
--- Returns true if the filename represents a C/C++ source code file. This check
--- is used to prevent passing non-code files to the compiler in makefiles. It is
--- not foolproof, but it has held up well. I'm open to better suggestions.
+-- Returns true if the filename represents various source languages.
 --
 
+	function path.isasmfile(fname)
+		return path.hasextension(fname, { ".s" })
+	end
+
 	function path.iscfile(fname)
-		return path.hasextension(fname, { ".c", ".s", ".m" })
+		return path.hasextension(fname, { ".c" })
+			or path.isasmfile(fname)	-- is this really right?
+			or path.isobjcfile(fname)	-- there is code that depends on this behaviour, which would need to change
 	end
 
 	function path.iscppfile(fname)
-		return path.hasextension(fname, { ".cc", ".cpp", ".cxx", ".c", ".s", ".m", ".mm" })
+		return path.hasextension(fname, { ".cc", ".cpp", ".cxx" })
```

Comment:
I don't disagree, it's completely valid (I tested it on Windows lol) and it seems to be used by things. I think @starkos has a better idea though.